packages/cloud_firestore/cloud_firestore_platform_interface/lib/cloud_firestore_platform_interface.dart

```diff
@@ -0,0 +1,219 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:async';
+
+import 'package:meta/meta.dart' show required, visibleForTesting;
+
+import 'src/method_channel_cloud_firestore.dart';
+
+import 'src/types.dart';
+
+export 'src/types.dart';
+
+/// The interface that implementations of `cloud_firestore` must extend.
+///
+/// Platform implementations should extend this class rather than implement it
+/// as `cloud_firestore` does not consider newly added methods to be breaking
+/// changes. Extending this class (using `extends`) ensures that the subclass
+/// will get the default implementation, while platform implementations that
+/// `implements` this interface will be broken by newly added
+/// [CloudFirestorePlatform] methods.
+abstract class CloudFirestorePlatform {
+  /// Only mock implementations should set this to `true`.
+  ///
+  /// Mockito mocks implement this class with `implements` which is forbidden
+  /// (see class docs). This property provides a backdoor for mocks to skip the
+  /// verification that the class isn't implemented with `implements`.
+  @visibleForTesting
+  bool get isMock => false;
```

Comment:
Fine for now but note that this will change soon, see https://github.com/flutter/plugins/pull/2348

Reply:
That's a great change, I'll keep an eye on that. (paging @hterkelsen who's also interested in that :P)

Reply:
(I migrated the new version to use this!)